control-plane/pkg/apis/messaging/v1beta1/kafka_channel_validation.go

```diff
@@ -113,3 +115,44 @@ func (kc *KafkaChannel) CheckImmutableFields(_ context.Context, original *KafkaC
 
 	return nil
 }
+
+func (kc *KafkaChannel) CheckSubscribersChangeAllowed(ctx context.Context, original *KafkaChannel) *apis.FieldError {
```

Comment:
@creydr I think you need to call this function in `Validate` - see here: https://github.com/knative/eventing/blob/d989ca732fad3701fef677f80594504512a39ee5/pkg/apis/messaging/v1/in_memory_channel_validation.go#L45-L49

Reply:
ah ***. Forgot. Thanks and good catch. Updated it